### ISSUE TITLE
New lint: `collapsible_str_replace`

### ISSUE THREAD

**Issue by camsteffen** [rust-lang/rust-clippy#6651]
### What it does

Checks for consecutive calls to `str::replace` (2 or more) that can be collapsed into a single call.

### Categories (optional)

- Kind: perf

*What is the advantage of the recommended code over the original code*

Faster since the string is only scanned once. Also less repetitive code.

### Drawbacks

None.

### Example

```rust
"hesuo worpd"
    .replace('s', "l")
    .replace("u", "l")
    .replace('p', "l")
```

Could be written as:

```rust
"hesuo worpd".replace(|c| matches!(c, 's' | 'u' | 'p'), "l")
```

Using `matches!` is potentially faster than a slice of chars (`replace(&['s', 'u', 'p'], "l")`). But if any of the chars are variables, the lint can fall back to suggesting a slice of chars.

**Comment by nahuakang** [comment]
This actually looks really fun. I'd like to move on to this issue :raised_hands:

**Comment by xFrednet** [comment]
@nahuakang could you maybe assign your self by commenting `@rustbot claim`? This makes it easier to see that someone is working on this issue. You can unassign yourself at anytime if you want to stop working on this :upside_down_face:. It's nice to see you around, keep up the good work :+1:
